File: py/src/ai/h2o/sparkling/SharedBackendConf.py
#
# Licensed to the Apache Software Foundation (ASF) under one or more
# contributor license agreements.  See the NOTICE file distributed with
# this work for additional information regarding copyright ownership.
# The ASF licenses this file to You under the Apache License, Version 2.0
# (the "License"); you may not use this file except in compliance with
# the License.  You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

from ai.h2o.sparkling.SharedBackendConfUtils import SharedBackendConfUtils
import warnings

class SharedBackendConf(SharedBackendConfUtils):

    #
    # Getters
    #

    def backendClusterMode(self):
        return self._jconf.backendClusterMode()

    def cloudName(self):
        return self._get_option(self._jconf.cloudName())

    def nthreads(self):
        return self._jconf.nthreads()

    def isH2OReplEnabled(self):
        return self._jconf.isH2OReplEnabled()

    def scalaIntDefaultNum(self):
        return self._jconf.scalaIntDefaultNum()

    def isClusterTopologyListenerEnabled(self):
        return self._jconf.isClusterTopologyListenerEnabled()

    def isSparkVersionCheckEnabled(self):
        return self._jconf.isSparkVersionCheckEnabled()

    def isFailOnUnsupportedSparkParamEnabled(self):
        return self._jconf.isFailOnUnsupportedSparkParamEnabled()

    def jks(self):
        return self._get_option(self._jconf.jks())

    def jksPass(self):
        return self._get_option(self._jconf.jksPass())

    def jksAlias(self):
        return self._get_option(self._jconf.jksAlias())

    def hashLogin(self):
        return self._jconf.hashLogin()

    def ldapLogin(self):
        return self._jconf.ldapLogin()

    def kerberosLogin(self):
        return self._jconf.kerberosLogin()

    def loginConf(self):
        return self._get_option(self._jconf.loginConf())

    def userName(self):
        return self._get_option(self._jconf.userName())

    def password(self):
        return self._get_option(self._jconf.password())

    def sslConf(self):
        return self._get_option(self._jconf.sslConf())

    def autoFlowSsl(self):
        return self._jconf.autoFlowSsl()

    def h2oNodeLogLevel(self):
        warnings.warn("The method 'h2oNodeLogLevel' is deprecated and will be removed in 3.34. Use 'logLevel' instead!")
        return self._jconf.logLevel()

    def logLevel(self):
        return self._jconf.logLevel()

    def h2oNodeLogDir(self):
        warnings.warn("The method 'h2oNodeLogDir' is deprecated and will be removed in 3.34. Use 'logDir' instead!")
        return self.logDir()

    def logDir(self):
        return self._get_option(self._jconf.logDir())

    def backendHeartbeatInterval(self):
        return self._jconf.backendHeartbeatInterval()

    def cloudTimeout(self):
        return self._jconf.cloudTimeout()

    def nodeNetworkMask(self):
        return self._get_option(self._jconf.nodeNetworkMask())

    def stacktraceCollectorInterval(self):
        return self._jconf.stacktraceCollectorInterval()

    def contextPath(self):
        return self._get_option(self._jconf.contextPath())

    def flowScalaCellAsync(self):
        return self._jconf.flowScalaCellAsync()

    def maxParallelScalaCellJobs(self):
        return self._jconf.maxParallelScalaCellJobs()

    def internalPortOffset(self):
        return self._jconf.internalPortOffset()

    def mojoDestroyTimeout(self):
        return self._jconf.mojoDestroyTimeout()

    def nodeBasePort(self):
        warnings.warn("The method 'nodeBasePort' is deprecated and will be removed in 3.34. Use 'basePort' instead!")
        return self.basePort()

    def basePort(self):
        return self._jconf.basePort()

    def nodeExtraProperties(self):
        warnings.warn("The method 'nodeExtraProperties' is deprecated and will be removed in 3.34. Use 'extraProperties' instead!")
        return self.extraProperties()

    def extraProperties(self):
        return self._get_option(self._jconf.extraProperties())

    def flowExtraHttpHeaders(self):
        return self._get_option(self._jconf.flowExtraHttpHeaders())

    def isInternalSecureConnectionsEnabled(self):
        return self._jconf.isInternalSecureConnectionsEnabled()

    def isInsecureXGBoostAllowed(self):
        return self._jconf.isInsecureXGBoostAllowed()

    def flowDir(self):
        return self._get_option(self._jconf.flowDir())

    def clientIp(self):
        return self._get_option(self._jconf.clientIp())

    def clientIcedDir(self):
        warnings.warn("The method 'clientIcedDir' is deprecated and will be removed in 3.34. Use 'icedDir' instead!")
        return self.icedDir()

    def h2oClientLogLevel(self):
        warnings.warn("The method 'h2oClientLogLevel' is deprecated and will be removed in 3.34. Use 'logLevel' instead!")
        return self._jconf.logLevel()

    def h2oClientLogDir(self):
        warnings.warn("The method 'h2oClientLogDir' is deprecated and will be removed in 3.34. Use 'logDir' instead!")
        return self.logDir()

    def clientBasePort(self):
        warnings.warn("The method 'clientBasePort' is deprecated and will be removed in 3.34. Use 'basePort' instead!")
        return self.basePort()

    def clientWebPort(self):
        return self._jconf.clientWebPort()

    def clientVerboseOutput(self):
        return self._jconf.clientVerboseOutput()

    def clientNetworkMask(self):
        return self._get_option(self._jconf.clientNetworkMask())

    def clientFlowBaseurlOverride(self):
        return self._get_option(self._jconf.clientFlowBaseurlOverride())

    def clientExtraProperties(self):
        warnings.warn("The method 'clientExtraProperties' is deprecated and will be removed in 3.34. Use 'extraProperties' instead!")
        return self.extraProperties()

    def runsInExternalClusterMode(self):
        return self._jconf.runsInExternalClusterMode()

    def runsInInternalClusterMode(self):
        return self._jconf.runsInInternalClusterMode()

    def clientCheckRetryTimeout(self):
        return self._jconf.clientCheckRetryTimeout()

    def verifySslCertificates(self):
        return self._jconf.verifySslCertificates()

    def isHiveSupportEnabled(self):
        return self._jconf.isHiveSupportEnabled()

    def hiveHost(self):
        return self._get_option(self._jconf.hiveHost())

    def hivePrincipal(self):
        return self._get_option(self._jconf.hivePrincipal())

    def hiveJdbcUrlPattern(self):
        return self._get_option(self._jconf.hiveJdbcUrlPattern())

    def hiveToken(self):
        return self._get_option(self._jconf.hiveToken())

    def icedDir(self):
        return self._get_option(self._jconf.icedDir())
    #
    # Setters
    #

    def setInternalClusterMode(self):
        self._jconf.setInternalClusterMode()
        return self

    def setExternalClusterMode(self):
        self._jconf.setExternalClusterMode()
        return self

    def setCloudName(self, cloudName):
        self._jconf.setCloudName(cloudName)
        return self

    def setNthreads(self, nthreads):
        self._jconf.setNthreads(nthreads)
        return self

    def setReplEnabled(self):
        self._jconf.setReplEnabled()
        return self

    def setReplDisabled(self):
        self._jconf.setReplDisabled()
        return self

    def setDefaultNumReplSessions(self, numSessions):
        self._jconf.setDefaultNumReplSessions(numSessions)
        return self

    def setClusterTopologyListenerEnabled(self):
        self._jconf.setClusterTopologyListenerEnabled()
        return self

    def setClusterTopologyListenerDisabled(self):
        self._jconf.setClusterTopologyListenerDisabled()
        return self

    def setSparkVersionCheckEnabled(self):
        self._jconf.setSparkVersionCheckEnabled()
        return self

    def setSparkVersionCheckDisabled(self):
        self._jconf.setSparkVersionCheckDisabled()
        return self

    def setFailOnUnsupportedSparkParamEnabled(self):
        self._jconf.setFailOnUnsupportedSparkParamEnabled()
        return self

    def setFailOnUnsupportedSparkParamDisabled(self):
        self._jconf.setFailOnUnsupportedSparkParamDisabled()
        return self

    def setJks(self, path):
        self._jconf.setJks(path)
        return self

    def setJksPass(self, password):
        self._jconf.setJksPass(password)
        return self

    def setJksAlias(self, alias):
        self._jconf.setJksAlias(alias)
        return self

    def setHashLoginEnabled(self):
        self._jconf.setHashLoginEnabled()
        return self

    def setHashLoginDisabled(self):
        self._jconf.setHashLoginDisabled()
        return self

    def setLdapLoginEnabled(self):
        self._jconf.setLdapLoginEnabled()
        return self

    def setLdapLoginDisabled(self):
        self._jconf.setLdapLoginDisabled()
        return self

    def setKerberosLoginEnabled(self):
        self._jconf.setKerberosLoginEnabled()
        return self

    def setKerberosLoginDisabled(self):
        self._jconf.setKerberosLoginDisabled()
        return self

    def setLoginConf(self, filePath):
        self._jconf.setLoginConf(filePath)
        return self

    def setUserName(self, username):
        self._jconf.setUserName(username)
        return self

    def setPassword(self, password):
        self._jconf.setPassword(password)
        return self

    def setSslConf(self, path):
        self._jconf.setSslConf(path)
        return self

    def setAutoFlowSslEnabled(self):
        self._jconf.setAutoFlowSslEnabled()
        return self

    def setAutoFlowSslDisabled(self):
        self._jconf.setAutoFlowSslDisabled()
        return self

    def setH2ONodeLogLevel(self, level):
        warnings.warn("The method 'setH2ONodeLogLevel' is deprecated and will be removed in 3.34. Use 'setLogLevel' instead!")
        return self.setLogLevel(level)

    def setLogLevel(self, level):
        self._jconf.setLogLevel(level)
        return self

    def setH2ONodeLogDir(self, dir):
        warnings.warn("The method 'setH2ONodeLogDir' is deprecated and will be removed in 3.34. Use 'setLogDir' instead!")
        return self.setLogDir(dir)

    def setLogDir(self, dir):
        self._jconf.setLogDir(dir)
        return self

    def setBackendHeartbeatInterval(self, interval):
        self._jconf.setBackendHeartbeatInterval(interval)
        return self

    def setCloudTimeout(self, timeout):
        self._jconf.setCloudTimeout(timeout)
        return self

    def setNodeNetworkMask(self, mask):
        self._jconf.setNodeNetworkMask(mask)
        return self

    def setStacktraceCollectorInterval(self, interval):
        self._jconf.setStacktraceCollectorInterval(interval)
        return self

    def setContextPath(self, contextPath):
        self._jconf.setContextPath(contextPath)
        return self

    def setFlowScalaCellAsyncEnabled(self):
        self._jconf.setFlowScalaCellAsyncEnabled()
        return self

    def setFlowScalaCellAsyncDisabled(self):
        self._jconf.setFlowScalaCellAsyncDisabled()
        return self

    def setMaxParallelScalaCellJobs(self, limit):
        self._jconf.setMaxParallelScalaCellJobs(limit)
        return self

    def setInternalPortOffset(self, offset):
        self._jconf.setInternalPortOffset(offset)
        return self

    def setNodeBasePort(self, port):
        warnings.warn("The method 'setNodeBasePort' is deprecated and will be removed in 3.34. Use 'setBasePort' instead!")
        return self.setBasePort(port)

    def setBasePort(self, port):
        self._jconf.setBasePort(port)
        return self

    def setMojoDestroyTimeout(self, timeoutInMilliseconds):
        self._jconf.setMojoDestroyTimeout(timeoutInMilliseconds)
        return self

    def setNodeExtraProperties(self, extraProperties):
        warnings.warn("The method 'setNodeExtraProperties' is deprecated and will be removed in 3.34. Use 'setExtraProperties' instead!")
        return self.setExtraProperties(extraProperties)

    def setExtraProperties(self, extraProperties):
        self._jconf.setExtraProperties(extraProperties)
        return self

    def setFlowExtraHttpHeaders(self, headers):
        self._jconf.setFlowExtraHttpHeaders(headers)
        return self

    def setInternalSecureConnectionsEnabled(self):
        self._jconf.setInternalSecureConnectionsEnabled()
        return self

    def setInternalSecureConnectionsDisabled(self):
        self._jconf.setInternalSecureConnectionsDisabled()
        return self

    def setInsecureXGBoostAllowed(self):
        self._jconf.setInsecureXGBoostAllowed()
        return self

    def setInsecureXGBoostDenied(self):
        self._jconf.setInsecureXGBoostDenied()
        return self

    def setFlowDir(self, dir):
        self._jconf.setFlowDir(dir)
        return self

    def setClientIp(self, ip):
        self._jconf.setClientIp(ip)
        return self

    def setClientIcedDir(self, dir):
        warnings.warn("The method 'setClientIcedDir' is deprecated and will be removed in 3.34. Use 'setIcedDir' instead!")
        return self.setIcedDir(dir)

    def setH2OClientLogLevel(self, level):
        warnings.warn("The method 'setH2OClientLogLevel' is deprecated and will be removed in 3.34. Use 'setLogLevel' instead!")
        return self.setLogLevel(level)

    def setH2OClientLogDir(self, dir):
        warnings.warn("The method 'setH2ONodeLogDir' is deprecated and will be removed in 3.34. Use 'setLogDir' instead!")
        return self.setLogDir(dir)

    def setClientBasePort(self, basePort):
        warnings.warn("The method 'setClientBasePort' is deprecated and will be removed in 3.34. Use 'setBasePort' instead!")
        return self.setBasePort(basePort)

    def setClientWebPort(self, port):
        self._jconf.setClientWebPort(port)
        return self

    def setClientVerboseEnabled(self):
        self._jconf.setClientVerboseEnabled()
        return self

    def setClientVerboseDisabled(self):
        self._jconf.setClientVerboseDisabled()
        return self

    def setClientNetworkMask(self, mask):
        self._jconf.setClientNetworkMask(mask)
        return self

    def setClientFlowBaseurlOverride(self, baseUrl):
        self._jconf.setClientFlowBaseurlOverride(baseUrl)
        return self

    def setClientCheckRetryTimeout(self, timeout):
        self._jconf.setClientCheckRetryTimeout(timeout)
        return self

    def setClientExtraProperties(self, extraProperties):
        warnings.warn("The method 'setClientExtraProperties' is deprecated and will be removed in 3.34. Use 'setExtraProperties' instead!")
        return self.setExtraProperties(extraProperties)

    def setVerifySslCertificates(self, verify):
        self._jconf.setVerifySslCertificates(verify)
        return self

    def setHiveSupportEnabled(self):
        self._jconf.setHiveSupportEnabled()
        return self

    def setHiveSupportDisabled(self):
        self._jconf.setHiveSupportDisabled()
        return self

    def setHiveHost(self, host):
        self._jconf.setHiveHost(host)
        return self

    def setHivePrincipal(self, principal):
        self._jconf.setHivePrincipal(principal)
        return self

    def setHiveJdbcUrlPattern(self, pattern):
        self._jconf.setHiveJdbcUrlPattern(pattern)
        return self

    def setHiveToken(self, token):
        self._jconf.setHiveToken(token)
        return self

    def setIcedDir(self, dir):
        self._jconf.setIcedDir(dir)
        return self

File: r/src/R/ai/h2o/sparkling/SharedBackendConf.R
#
# Licensed to the Apache Software Foundation (ASF) under one or more
# contributor license agreements.  See the NOTICE file distributed with
# this work for additional information regarding copyright ownership.
# The ASF licenses this file to You under the Apache License, Version 2.0
# (the "License"); you may not use this file except in compliance with
# the License.  You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

if (!exists("ConfUtils.getOption", mode = "function")) source(file.path("R", "ConfUtils.R"))

#' @export SharedBackendConf
SharedBackendConf <- setRefClass("SharedBackendConf", methods = list(

#
# Getters
#
    backendClusterMode = function() { invoke(jconf, "backendClusterMode") },

    cloudName = function() { ConfUtils.getOption(invoke(jconf, "cloudName")) },

    nthreads = function() { invoke(jconf, "nthreads") },

    isH2OReplEnabled = function() { invoke(jconf, "isH2OReplEnabled") },

    scalaIntDefaultNum = function() { invoke(jconf, "scalaIntDefaultNum") },

    isClusterTopologyListenerEnabled = function() { invoke(jconf, "isClusterTopologyListenerEnabled") },

    isSparkVersionCheckEnabled = function() { invoke(jconf, "isSparkVersionCheckEnabled") },

    isFailOnUnsupportedSparkParamEnabled = function() { invoke(jconf, "isFailOnUnsupportedSparkParamEnabled") },

    jks = function() { ConfUtils.getOption(invoke(jconf, "jks")) },

    jksPass = function() { ConfUtils.getOption(invoke(jconf, "jksPass")) },

    jksAlias = function() { ConfUtils.getOption(invoke(jconf, "jksAlias")) },

    hashLogin = function() { invoke(jconf, "hashLogin") },

    ldapLogin = function() { invoke(jconf, "ldapLogin") },

    kerberosLogin = function() { invoke(jconf, "kerberosLogin") },

    loginConf = function() { ConfUtils.getOption(invoke(jconf, "loginConf")) },

    userName = function() { ConfUtils.getOption(invoke(jconf, "userName")) },

    password = function() { ConfUtils.getOption(invoke(jconf, "password")) },

    sslConf = function() { ConfUtils.getOption(invoke(jconf, "sslConf")) },

    autoFlowSsl = function() { invoke(jconf, "autoFlowSsl") },

    h2oNodeLogLevel = function() { invoke(jconf, "h2oNodeLogLevel") },

    h2oNodeLogDir = function() { ConfUtils.getOption(invoke(jconf, "h2oNodeLogDir")) },

    backendHeartbeatInterval = function() { invoke(jconf, "backendHeartbeatInterval") },

    cloudTimeout = function() { invoke(jconf, "cloudTimeout") },

    nodeNetworkMask = function() { ConfUtils.getOption(invoke(jconf, "nodeNetworkMask")) },

    stacktraceCollectorInterval = function() { invoke(jconf, "stacktraceCollectorInterval") },

    contextPath = function() { ConfUtils.getOption(invoke(jconf, "contextPath")) },

    flowScalaCellAsync = function() { invoke(jconf, "flowScalaCellAsync") },

    maxParallelScalaCellJobs = function() { invoke(jconf, "maxParallelScalaCellJobs") },

    internalPortOffset = function() { invoke(jconf, "internalPortOffset") },

    mojoDestroyTimeout = function() { invoke(jconf, "mojoDestroyTimeout") },

    nodeBasePort = function() { invoke(jconf, "nodeBasePort") },

    nodeExtraProperties = function() { ConfUtils.getOption(invoke(jconf, "nodeExtraProperties")) },

    flowExtraHttpHeaders = function() { ConfUtils.getOption(invoke(jconf, "flowExtraHttpHeaders")) },

    isInternalSecureConnectionsEnabled = function() { invoke(jconf, "isInternalSecureConnectionsEnabled") },

    isSparkTimeZoneFollowed = function() { invoke(jconf, "isSparkTimeZoneFollowed") },

    flowDir = function() { ConfUtils.getOption(invoke(jconf, "flowDir")) },

    clientIp = function() { ConfUtils.getOption(invoke(jconf, "clientIp")) },

    clientIcedDir = function() { ConfUtils.getOption(invoke(jconf, "clientIcedDir")) },

    h2oClientLogLevel = function() { invoke(jconf, "h2oClientLogLevel") },

    h2oClientLogDir = function() { ConfUtils.getOption(invoke(jconf, "h2oClientLogDir")) },

    clientBasePort = function() { invoke(jconf, "clientBasePort") },

    clientWebPort = function() { invoke(jconf, "clientWebPort") },

    clientVerboseOutput = function() { invoke(jconf, "clientVerboseOutput") },

    clientNetworkMask = function() { ConfUtils.getOption(invoke(jconf, "clientNetworkMask")) },

    clientFlowBaseurlOverride = function() { ConfUtils.getOption(invoke(jconf, "clientFlowBaseurlOverride")) },

    clientExtraProperties = function() { ConfUtils.getOption(invoke(jconf, "clientExtraProperties")) },

    runsInExternalClusterMode = function() { invoke(jconf, "runsInExternalClusterMode") },

    runsInInternalClusterMode = function() { invoke(jconf, "runsInInternalClusterMode") },

    clientCheckRetryTimeout = function() { invoke(jconf, "clientCheckRetryTimeout") },

    verifySslCertificates = function() { invoke(jconf, "verifySslCertificates") },

#
# Setters
#
    setInternalClusterMode = function() { invoke(jconf, "setInternalClusterMode"); .self },

    setExternalClusterMode = function() { invoke(jconf, "setExternalClusterMode"); .self },

    setCloudName = function(cloudName) { invoke(jconf, "setCloudName", cloudName); .self },

    setNthreads = function(nthreads) { invoke(jconf, "setNthreads", as.integer(nthreads)); .self },

    setReplEnabled = function() { invoke(jconf, "setReplEnabled"); .self },

    setReplDisabled = function() { invoke(jconf, "setReplDisabled"); .self },

    setDefaultNumReplSessions = function(numSessions) { invoke(jconf, "setDefaultNumReplSessions", as.integer(numSessions)); .self },

    setClusterTopologyListenerEnabled = function() { invoke(jconf, "setClusterTopologyListenerEnabled"); .self },

    setClusterTopologyListenerDisabled = function() { invoke(jconf, "setClusterTopologyListenerDisabled"); .self },

    setSparkVersionCheckEnabled = function() { invoke(jconf, "setSparkVersionCheckEnabled"); .self },

    setSparkVersionCheckDisabled = function() { invoke(jconf, "setSparkVersionCheckDisabled"); .self },

    setFailOnUnsupportedSparkParamEnabled = function() { invoke(jconf, "setFailOnUnsupportedSparkParamEnabled"); .self },

    setFailOnUnsupportedSparkParamDisabled = function() { invoke(jconf, "setFailOnUnsupportedSparkParamDisabled"); .self },

    setJks = function(path) { invoke(jconf, "setJks", path); .self },

    setJksPass = function(password) { invoke(jconf, "setJksPass", password); .self },

    setJksAlias = function(alias) { invoke(jconf, "setJksAlias", alias); .self },

    setHashLoginEnabled = function() { invoke(jconf, "setHashLoginEnabled"); .self },

    setHashLoginDisabled = function() { invoke(jconf, "setHashLoginDisabled"); .self },

    setLdapLoginEnabled = function() { invoke(jconf, "setLdapLoginEnabled"); .self },

    setLdapLoginDisabled = function() { invoke(jconf, "setLdapLoginDisabled"); .self },

    setKerberosLoginEnabled = function() { invoke(jconf, "setKerberosLoginEnabled"); .self },

    setKerberosLoginDisabled = function() { invoke(jconf, "setKerberosLoginDisabled"); .self },

    setLoginConf = function(filePath) { invoke(jconf, "setLoginConf", filePath); .self },

    setUserName = function(username) { invoke(jconf, "setUserName", username); .self },

    setPassword = function(password) { invoke(jconf, "setPassword", password); .self },

    setSslConf = function(path) { invoke(jconf, "setSslConf", path); .self },

    setAutoFlowSslEnabled = function() { invoke(jconf, "setAutoFlowSslEnabled"); .self },

    setAutoFlowSslDisabled = function() { invoke(jconf, "setAutoFlowSslDisabled"); .self },

    setH2ONodeLogLevel = function(level) { invoke(jconf, "setH2ONodeLogLevel", level); .self },

    setH2ONodeLogDir = function(dir) { invoke(jconf, "setH2ONodeLogDir", dir); .self },

    setBackendHeartbeatInterval = function(interval) { invoke(jconf, "setBackendHeartbeatInterval", as.integer(interval)); .self },

    setCloudTimeout = function(timeout) { invoke(jconf, "setCloudTimeout", as.integer(timeout)); .self },

    setNodeNetworkMask = function(mask) { invoke(jconf, "setNodeNetworkMask", mask); .self },

    setStacktraceCollectorInterval = function(interval) { invoke(jconf, "setStacktraceCollectorInterval", as.integer(interval)); .self },

    setContextPath = function(contextPath) { invoke(jconf, "setContextPath", contextPath); .self },

    setFlowScalaCellAsyncEnabled = function() { invoke(jconf, "setFlowScalaCellAsyncEnabled"); .self },

    setFlowScalaCellAsyncDisabled = function() { invoke(jconf, "setFlowScalaCellAsyncDisabled"); .self },

    setMaxParallelScalaCellJobs = function(limit) { invoke(jconf, "setMaxParallelScalaCellJobs", as.integer(limit)); .self },

    setInternalPortOffset = function(offset) { invoke(jconf, "setInternalPortOffset", as.integer(offset)); .self },

    setNodeBasePort = function(port) { invoke(jconf, "setNodeBasePort", as.integer(port)); .self },

    setMojoDestroyTimeout = function(timeoutInMilliseconds) { invoke(jconf, "setMojoDestroyTimeout", as.integer(timeoutInMilliseconds)); .self },

    setNodeExtraProperties = function(extraProperties) { invoke(jconf, "setNodeExtraProperties", extraProperties); .self },

    setFlowExtraHttpHeaders = function(headers) { invoke(jconf, "setFlowExtraHttpHeaders", headers); .self },

    setInternalSecureConnectionsEnabled = function() { invoke(jconf, "setInternalSecureConnectionsEnabled"); .self },

    setInternalSecureConnectionsDisabled = function() { invoke(jconf, "setInternalSecureConnectionsDisabled"); .self },

    setSparkTimeZoneFollowingEnabled = function() { invoke(jconf, "setSparkTimeZoneFollowingEnabled"); .self },

    setSparkTimeZoneFollowingDisabled = function() { invoke(jconf, "setSparkTimeZoneFollowingDisabled"); .self },

    setFlowDir = function(dir) { invoke(jconf, "setFlowDir", dir); .self },

    setClientIp = function(ip) { invoke(jconf, "setClientIp", ip); .self },

    setClientIcedDir = function(icedDir) { invoke(jconf, "setClientIcedDir", icedDir); .self },

    setH2OClientLogLevel = function(level) { invoke(jconf, "setH2OClientLogLevel", level); .self },

    setH2OClientLogDir = function(dir) { invoke(jconf, "setH2OClientLogDir", dir); .self },

    setClientBasePort = function(basePort) { invoke(jconf, "setClientBasePort", as.integer(basePort)); .self },

    setClientWebPort = function(port) { invoke(jconf, "setClientWebPort", as.integer(port)); .self },

    setClientVerboseEnabled = function() { invoke(jconf, "setClientVerboseEnabled"); .self },

    setClientVerboseDisabled = function() { invoke(jconf, "setClientVerboseDisabled"); .self },

    setClientNetworkMask = function(mask) { invoke(jconf, "setClientNetworkMask", mask); .self },

    setClientFlowBaseurlOverride = function(baseUrl) { invoke(jconf, "setClientFlowBaseurlOverride", baseUrl); .self },

    setClientCheckRetryTimeout = function(timeout) { invoke(jconf, "setClientCheckRetryTimeout", as.integer(timeout)); .self },

    setClientExtraProperties = function(extraProperties) { invoke(jconf, "setClientExtraProperties", extraProperties); .self },

    setVerifySslCertificates = function(verify) { invoke(jconf, "setVerifySslCertificates", verify); .self }
))
